watchdog.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+/**
+ * This is a watch dog timer.  Think of it as a ticking
+ * timebomb which will throw an explosion when it hasn't
+ * been stopped or touched in `maxTime` seconds.  The
+ * `WatchDog` will throw an `Error` with `msg` if the
+ * timer is allowed to expire
+ */
+function WatchDog(maxTime, action) {
+  this.maxTime = maxTime;
+  if (typeof action === 'function') {
+    this.action = action;
+  } else {
+    this.action = function() {
+      throw new Error(action);
```

Comment:
I've modified the WatchDog to be an eventemitter.  In `provisioner/provision.js`, though, I'm throwing an error still on the `expired` event.  Crashing node in this case is the desired behaviour, unless you know of a way that I can 'cancel' a promise from outside the promise...